client/schema_field_description.go

```diff
@@ -169,3 +280,74 @@ var FieldKindStringToEnumMapping = map[string]FieldKind{
 func (f SchemaFieldDescription) IsRelation() bool {
 	return f.RelationName != ""
 }
+
+// schemaFieldDescription is a private type used to facilitate the unmarshalling
+// of json to a [SchemaFieldDescription].
+type schemaFieldDescription struct {
+	Name              string
+	RelationName      string
+	Typ               CType
+	IsPrimaryRelation bool
+
+	// Properties below this line are unmarshalled using custom logic in [UnmarshalJSON]
+	Kind json.RawMessage
+}
+
+func (f *SchemaFieldDescription) UnmarshalJSON(bytes []byte) error {
+	var descMap schemaFieldDescription
+	err := json.Unmarshal(bytes, &descMap)
+	if err != nil {
+		return err
+	}
+
+	f.Name = descMap.Name
+	f.RelationName = descMap.RelationName
+	f.Typ = descMap.Typ
+	f.IsPrimaryRelation = descMap.IsPrimaryRelation
+
+	if len(descMap.Kind) == 0 {
+		f.Kind = FieldKind_None
+		return nil
+	}
+
+	if descMap.Kind[0] != '"' {
+		// If the Kind is not represented by a string, assume try to parse it to an int, as
+		// that is the only other type we support.
+		var intKind uint8
+		err := json.Unmarshal(descMap.Kind, &intKind)
+		if err != nil {
+			return err
+		}
+		switch intKind {
+		case uint8(FieldKind_BOOL_ARRAY), uint8(FieldKind_INT_ARRAY), uint8(FieldKind_FLOAT_ARRAY),
+			uint8(FieldKind_STRING_ARRAY), uint8(FieldKind_NILLABLE_BOOL_ARRAY), uint8(FieldKind_NILLABLE_INT_ARRAY),
+			uint8(FieldKind_NILLABLE_FLOAT_ARRAY), uint8(FieldKind_NILLABLE_STRING_ARRAY):
+			f.Kind = ScalarArrayKind(intKind)
+		default:
+			f.Kind = ScalarKind(intKind)
```

Comment:
thought: This as the potential to be quite annoying to maintain as we update out list of supported array types. I'm wondering if we should take the hit now and change the assigned number ranges to something like:
- [0:100] -> Scalar types
- [100:200] -> Array types

We wont be using the full range but at least if we ever need to do another switch statement it would be a much cleaner check.

Reply:
I dislike the current too, however I don't like the range option, and would probably prefer a custom encoding: First bit for nillable, second bit for array, then the scalar id.

As a proper solution would require a bit of thought and an agreed design, I'd prefer if this conversation was out of scope here.